qc_utils/test/test_qcmetric.py

```diff
@@ -70,6 +71,18 @@ def test_metric_to_ordered_dict(obj_d):
     assert obj_d.to_ordered_dict() == OrderedDict([("d", OrderedDict([("a", "b")]))])
 
 
+def test_QCMetric_save(mocker, obj_a1):
+    """
+    `json.dump` splits the writes so we need to gather them from the mocked calls. The
+    first two calls are the `open()` and context manager's `__enter__` and the last call
+    is the context manager `__exit__`, we need to skip them.
+    """
+    mocker.patch("builtins.open", mocker.mock_open())
+    obj_a1.save("foo.json")
+    written = "".join([i[1][0] for i in builtins.open.mock_calls[2:-1]])
```

Comment:
I think this is going too much into detail of how json.dump works and even what it writes. I think the thing we want to test here is that json.dump gets called once with the correct parameters. This way we test behavior and not some implementation (for example in case at some point we want to parametrize the writing method to write something else, like yaml or whatever we can basically use the same test).

Reply:
Yeah I agree, was kind of trying to write an integration test without writing an integration test.